src/main/java/org/opentripplanner/middleware/connecteddataplatform/ConnectedDataManager.java

```diff
@@ -232,21 +233,22 @@ private static Coordinates getPlaceCoordinates(
     }
 
     /**
-     * Extract the trip request that uses the most modes.
+     * A single trip query results in many calls from the UI to OTP covering different combinations of modes. The trip
+     * request to be included in the anonymous trip data must be the one which uses the most modes.
```

Comment:
This method could probably deleted in favor of retrieving the first request found for a batch. We don't use the number of modes involved for anything, do we?

Reply:
@binh-dam-ibigroup this method only exists so that the trip request with the most modes is serialized and uploaded to S3. This way the client can see the maximum number of modes that were used to produce the trip summaries. It is possible that I haven't understood this correctly, if so, let me know.